### PR TITLE
test: e2e test scaffolding

### LINKED ISSUE #120: Integrate nexus-e2e with Nexus GitHub workflow
https://github.com/gpBlockchain/nexus-e2e/blob/main/packages/e2e/README.md

### PR DESCRIPTION
**Pull request by homura**
# What Changed

Resolve #120 

Added end-to-end testing scaffolding. [Here](https://github.com/ckb-js/nexus/blob/simplify-e2e/e2e/tests/wallet-enable.test.ts) is an example showing how to add a test case

## Motivation

This PR is heavily inspired by #151, but with some simplification

- a simple `TestEnv` for smoothing out runtime differences between browser and nodejs
- one file, one scenario
- jest as reporter

## Change Type

Indicate the type of change your pull request is:

- [ ] `documentation`
- [ ] `patch`
- [ ] `minor`
- [ ] `major`

### CODE REVIEW

File: e2e/helpers/DefaultTestEnv.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      predicate: (page) => page.url().includes('notification.html'),
+    });
+    if (!notificationPage) {
+      throw new Error('Notification page not found');
+    }
+
+    return notificationPage;
+  }
+
+  async approveForEnable(page?: Page): Promise<void> {
```
Comment (IronLu233) [inline e2e/helpers/DefaultTestEnv.ts:168]: `approveSigning`, `rejectSigning` may be also needed.

it is very common when test the signing features.

File: e2e/helpers/DefaultTestEnv.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        }, payload);
+
+        if (this.options.autoApproveEnable && payload.method === 'wallet_enable') {
+          await this.approveForEnable();
+        }
+
+        return await res;
+      },
+
+      on: errors.unimplemented,
```
Comment (IronLu233) [inline e2e/helpers/DefaultTestEnv.ts:198]: A test case may needed `on` listener:
User request grant -> user close grant window -> expect receive grant rejected message

File: e2e/helpers/DefaultTestEnv.ts (hunk 1 above)
Comment (homura) [reply e2e/helpers/DefaultTestEnv.ts:168]: indeed, but my purpose to open this PR is to simplify #151, so it is not a complete implementation yet

Comment (IronLu233) [reply e2e/helpers/DefaultTestEnv.ts:168]: I see it's just a scaffolding. I will resolve it.

**Review by IronLu233 — APPROVED** [review]

**Pull request merged**